test/jdk/java/lang/StringBuffer/IndexOf.java

```diff
@@ -24,165 +24,388 @@
 /* @test
  * @bug 4162796 4162796
  * @summary Test indexOf and lastIndexOf
+ * @run main/othervm IndexOf
  * @key randomness
  */
 
+/* @test
+ * @bug 8320448
+ * @summary Test indexOf
+ * @requires vm.cpu.features ~= ".*avx2.*"
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions -Xcomp -XX:-TieredCompilation -XX:UseAVX=2 -XX:+UnlockDiagnosticVMOptions -XX:+EnableX86ECoreOpts IndexOf
+ * @key randomness
+ */
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
 import java.util.Random;
 
 public class IndexOf {
 
-    static Random generator = new Random();
-    private static boolean failure = false;
+  static Random generator = new Random();
```

Comment:
@RogerRiggs Would you have cycles to look at Scott's changes to this test? I suspect it will need to be re-structured, re-formatted, and commented to get into maintainable shape.

Reply:
I am going to revert my changes to this file as the test `jdk/java/lang/String/IndexOf.java` covers the code better.